### PR TITLE
Coordination Across Programming Models (OpenMP/MPI)

### PR DESCRIPTION
**Pull request by rhc54**
Signed-off-by: Ralph Castain <rhc@open-mpi.org>

### CODE REVIEW

File: RFC0017.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+        exit(active);
+    }
+
+    /* do whatever else */
+}
+```
+
+#### Sharing Resources
+
+## Protoype Implementation
```
Comment (rhc54) [inline RFC0017.md]: Section on OpenMP envars and how they would translate into this approach

File: RFC0017.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        fprintf(stderr, "Client ns %s rank %d: PMIx_Init failed: %d\n", myproc.nspace, myproc.rank, rc);
+        exit(0);
+    }
+    PMIX_INFO_FREE(info, 4);
+    fprintf(stderr, "Client ns %s rank %d: Running\n", myproc.nspace, myproc.rank);
+
+    /* register a handler specifically for notifying us if
+     * other programming models declare themselves */
+    active = -1;
+    PMIx_Register_event_handler(&code, 1, NULL, 0,
```
Comment (boehms) [inline RFC0017.md:135]: Is it possible to register muliple event handler for the same event?

Comment (rhc54) [reply RFC0017.md:135]: Not at the moment - however, that is being changed per the following RFC: https://github.com/pmix/RFCs/pull/18

**Comment by hppritcha** [comment]
It does look like one of the DOE mini-apps - snap would be a good guinea pig for testing the usefulness of this RFC.  It can use nested OpenMP, and also has ports to kokkos - although its unclear if this works.

https://github.com/lanl/SNAP

File: RFC0017.md (hunk 2 above)
Comment (hppritcha) [reply RFC0017.md:135]: we might need to have a more flexible way to control when a programming model's callback function is invoked.  Looking at section 3.2 of the OpenMP spec there are a number of calls that may lead to the OpenMP runtime adjusting resource requirements etc. that other program models might need to take in to account.  I'm thinking, for example, of ```omp_set_nested```.

**Comment by rhc54** [comment]
This has been "seconded" by ORNL.

**Pull request merged**